api/paidAction/index.js

```diff
@@ -30,75 +34,166 @@ export const paidActions = {
   TERRITORY_BILLING,
   TERRITORY_UNARCHIVE,
   DONATE,
+  TRANSFER,
   BUY_CREDITS
 }
 
+async function checkUser (userId, { models, tx, me }, minSats, minCreds) {
+  // count pending invoices and bail if we're over the limit
+  const pendingInvoices = await ((tx ?? models).invoice.count({
+    where: {
+      userId,
+      actionState: {
+        // not in a terminal state. Note: null isn't counted by prisma
+        notIn: PAID_ACTION_TERMINAL_STATES
+      }
+    }
+  }))
+
+  console.log('pending paid actions', pendingInvoices)
+  if (pendingInvoices >= MAX_PENDING_PAID_ACTIONS_PER_USER) {
+    throw new Error('you have too many pending paid actions, cancel some or wait for them to expire')
+  }
+
+  if (minSats != null && toPositiveBigInt(minSats) > (me?.msats ?? 0n)) {
+    throw new Error('you do not have enough sats to perform this action')
+  }
+  if (minCreds != null && toPositiveBigInt(minCreds) > (me?.mcredits ?? 0n)) {
+    throw new Error('you do not have enough credits to perform this action')
+  }
+}
+
 export default async function performPaidAction (actionType, args, context) {
   try {
-    const { me, models, forceFeeCredits } = context
-    const paidAction = paidActions[actionType]
-
+    context = { ...context }
     console.group('performPaidAction', actionType, args)
-
+    const paidAction = paidActions[actionType]
     if (!paidAction) {
-      throw new Error(`Invalid action type ${actionType}`)
+      throw new Error(`ynvalid action type ${actionType}`)
     }
 
-    if (!me && !paidAction.anonable) {
-      throw new Error('You must be logged in to perform this action')
-    }
+    context.me = context.me ? await context.models.user.findUnique({ where: { id: context.me.id } }) : undefined
+    context.actionAttempt = context.actionAttempt ?? 0
+    context.forceInternal = context.forceInternal ?? false
+    context.prioritizeInternal = context.prioritizeInternal ?? false
```

Comment:
Contrary to forceInternal, this will just prioritize reward sats and fee credits. 
This is used by the client side code after every sending wallet failed to pay every receiving wallet, so it will try to do an internal transaction, if this fails too they payment will fallback to the other methods in order, this should give another invoice that is then presented to the user with a qr code.